playground/hooks/pages/enroll-profile.ts

```diff
@@ -0,0 +1,125 @@
+          name: 'custom_bool',
+          type: 'boolean',
+          required: true,
+          options: [{
+            label: 'display',
+            value: {
+              type: 'object',
+              value: {
+                inputType: 'checkbox'
+              }
+            } as any
+          }]
+        });
+      }
+      // Add custom string "TIN"
+      if (!schema.find(f => f.name.includes('custom_string'))) {
+        schema.push({
+          label: 'Custom string',
+          name: 'custom_string',
+          type: 'string',
+          required: true,
+          minLength: 9,
+          maxLength: 9,
+        });
+      }
+      onSuccess(schema);
+    },
+  };
+};
+
+
+export const addHookForEnrollProfileForm = (oktaSignIn: OktaSignInAPIV3) => {
+  // Tip for Sign-in page code editor: 
+  //  Paste this code after `oktaSignIn = new OktaSignIn(config);`
+  oktaSignIn.afterTransform('enroll-profile', ({ formBag, loc }) => {
+    // Change title
+    const titleIndex = formBag.uischema.elements.findIndex(ele => ele.type === 'Title');
+    const title: TitleElement = formBag.uischema.elements[titleIndex] as TitleElement;
+    title.options.content = 'Register new profile';
+
+    // Change submit text
+    const submitIndex = formBag.uischema.elements.findIndex(ele =>
+      ele.type === 'Button' && (ele as ButtonElement).options.type === 'submit'
+    );
+    const submit: ButtonElement = formBag.uischema.elements[submitIndex] as ButtonElement;
+    submit.label = 'Register';
+
+    // Add custom description after title
+    const descr: DescriptionElement = {
+      type: 'Description',
+      contentType: 'subtitle',
+      options: {
+        variant: 'body1', // or 'subtitle1' or 'legend'
+        content: '<div class=\'my-enroll-description\'>Your<br />custom<br />description</div>'
+      },
+    };
+    formBag.uischema.elements.splice(titleIndex + 1, 0, descr);
+
+    // Customize custom string field (`custom_string`) - change label
+    const customString = formBag.uischema.elements.find<FieldElement>((ele): ele is FieldElement =>
+      ele.type === 'Field'
+      && (ele as FieldElement).options.type === 'string'
+      && (ele as FieldElement).options.inputMeta.name.includes('custom')
+    );
+    if (customString) {
+      customString.translations.find(t => t.name === 'label').value = loc('custom.field.tin.label');
+    }
+
+    // Customize custom boolean field (`custom_bool`) - add title, change label (with i18n support)
+    const customBoolIndex = formBag.uischema.elements.findIndex((ele): ele is FieldElement =>
+      ele.type === 'Field'
+      && (ele as FieldElement).options.type === 'boolean'
+      && (ele as FieldElement).options.inputMeta.name.includes('custom')
+    );
+    if (customBoolIndex != -1) {
+      const customBool = formBag.uischema.elements[customBoolIndex] as FieldElement;
+      customBool.translations.find(t => t.name === 'label').value = loc('custom.field.terms.label');
```

Comment:
Exposed `loc` util for `afterTransform` hook and used in this sample to demonstrate i18n capabilities. 
Translations are performed before `afterTransform` hook, so setting `i18nKey` in items of `translations` array would have no effect. 
Thoughts?